invenio/utils/deprecation.py

```diff
@@ -87,17 +89,49 @@ class RemovedInInvenio22Warning(DeprecationWarning):
     """Mark feature that will be removed in Invenio version 2.2."""
 
 
+def warn_with_traceback(message, category, filename, lineno, file=None,
+                        line=None):
+    """Log deprecation warnings to debug logger."""
+    # Format
+    message.args += 'Full backtrace in the logfile if DEBUG is enabled',
+    formatted_warning = warnings.formatwarning(message, category, filename, lineno, line)
+
+    # Warn
+    log = file if hasattr(file, 'write') else sys.stderr
+    log.write(formatted_warning)
+
+    # Debug
+    try:
+        current_app.logger.debug(formatted_warning)
+        current_app.logger.debug(''.join(traceback.format_stack()))
+    except RuntimeError:
+        # App not yet initialized; see no evil, hear no evil.
+        pass
+
+
+original_showwarning = warnings.showwarning
```

Comment:
Should this backup better made in line 132 instead? Otherwise the original state is not restored when `warnings.showwarning` is changed somewhere else after 112 is executed and therefore the decorator has unwanted side effects.